src/pylibftdi/device.py

```diff
@@ -321,7 +325,11 @@ def _open_device(self) -> int:
         """
         # FTDI vendor/product ids required here.
         res: int = -1
-        for usb_vid, usb_pid in itertools.product(USB_VID_LIST, USB_PID_LIST):
+        for usb_vid, usb_pid in itertools.product(
+            USB_VID_LIST + [self.vid], USB_PID_LIST + [self.pid]
+        ):
+            if usb_vid is None or usb_pid is None:
+                continue
```

Comment:
I think it would be better to have a provided VID / PID select *instead* of the default lists, rather than adding to them. In particular, if both a 'usual' VID/PID FTDI device and a custom one are present, a call to `Device(vid=X, pid=Y)` would still pick up the 'usual' rather than the custom one with this logic, which I think would be unexpected - and if *only* a 'usual' one was present it should probably **not** be selected if vid/pid are provided.

I think something like the following might be better to ensure it's 'either custom / or default lists (possibly modified as per docs)'

```python
vid_list = [self.vid] if self.vid is not None else USB_VID_LIST
pid_list = [self.pid] if self.pid is not None else USB_PID_LIST
for usb_vid, usb_pid In itertools.product(vid_list, pid_list):
    open_args = ...
```

Reply:
Good catch, I'll make this change.

Thank you so much for your responsiveness and receptiveness on this!